### PR TITLE
fix: handle /subpath-stuff/ when redirecting to subdomains

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Having a regex like /chinese/? meant that the desired paths /chinese/...
matched, but so did paths like /chinese-something-else.

This resulted in freecodecamp.org/chinese-traditional/ getting
redirected to chinese.freecodecamp.org/-traditional

While I was there, I also cleaned up all the other /?'s because I figured the same principle applied.

### CODE REVIEW

File: sites-enabled/10-www.freecodecamp.org.conf [hunk 1]
```diff
@@ -163,10 +163,10 @@ server {
   }
 
   # catch common subpath for langs that we have special configs for
-  location ~ ^/spanish/?(.*)$ {
+  location ~ ^/spanish/(.*)$ {
     return 302 https://www.freecodecamp.org/espanol/$1;
   }
-  location ~ ^/chinese/?(.*)$ {
+  location ~ ^/chinese/(.*)$ {
```
Comment (ShaunSHamilton) [inline sites-enabled/10-www.freecodecamp.org.conf:169]: Does this not mean that `freecodecamp.org/chinese` will **not** be redirected, when it should?

Comment (ojeytonwilliams) [reply sites-enabled/10-www.freecodecamp.org.conf:169]: Good question, I'll check.

Comment (ojeytonwilliams) [reply sites-enabled/10-www.freecodecamp.org.conf:169]: I'm glad I you asked... let me try again.

**Comment by ojeytonwilliams** [comment]
> LGTM +1 , but this does not include `/italian/` or Portuguese?

Nope, because there weren't any optional path segments for those languages.

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

**Comment by nhcarrigan** [comment]
> I suppose my question is why is Spanish special?

Because `/spanish` redirects to `/espanol`.
